src/cmdc/datasets/uscensus/data.py

```diff
@@ -0,0 +1,198 @@
+            A DataFrame with a column for each variable requested and
+            a column, `fips` which specifies the geographic information
+        """
+        # Fetch data
+        df = super(ACS_CMDC, self).get(self.cols, self.geo)
+
+        # Convert to fips representation
+        df = self._create_fips(df)
+
+        # Reshape into desired format
+        df = df.melt(id_vars="fips", var_name="census_id", value_name="value")
+
+        return df
+
+    def _var_insert_query(self, df, table_name, temp_name, pk):
+        _sql_var_insert = f"""
+        INSERT INTO uscensus.{table_name} (year, product, census_id, label)
+        SELECT year, product, census_id, label FROM {temp_name}
+        ON CONFLICT {pk} DO NOTHING;
+        """
+
+        return _sql_var_insert
+
+    def variable_get(self):
+        """
+        Fetches the variables for the specified ACS dataset
+
+        Returns
+        -------
+        all_variables : pd.DataFrame
+            A DataFrame with columns (year, product, census_id, label)
+            which matches the columsn in the `uscensus.acs_variables`
+            table
+        """
+        # Fetch variables json
+        variable_json = json.loads(
+            requests.get(self.dataset["c_variablesLink"]).text
+        )
+
+        # Load into DataFrame
+        all_variables = pd.DataFrame.from_dict(
+            variable_json["variables"]
+        ).T
+
+        # Only keep 'Estimate' variables
+        is_variable = all_variables["label"].str.contains("Estimate")
+        all_variables = all_variables.loc[is_variable, ["label"]].reset_index()
+
+        all_variables["year"] = self.dataset["c_vintage"]
+        all_variables["product"] = self.product
+        all_variables = all_variables.rename(columns={"index": "census_id"})
+
+        return all_variables
+
+    def _put(self, connstr, table="variable", df=None):
+        """
+        Inserts the variable data into the `uscensus.acs_variables`
+        table
+
+        Parameters
+        ----------
+        connstr : string
+            A postgres connection string that can be used by sqlalchemy
+        d_or_v : string
+            A string that denotes whether we're putting data or
+            variables into the database
+        df : pd.DataFrame
+            The DataFrame with all of the variables contained in a
+            particular data product
+        """
+        if df is None:
+            if table == "data":
+                df = self.variable_get()
+            elif table == "variable":
+                df = self.variable_get()
+
+        if table == "data":
+            tablename = self.data_table
+            pk = self.data_pk
+            _insert_cmd = self._data_insert_query
+        else:
+            tablename = self.variable_table
+            pk = self.variable_pk
+            _insert_cmd = self._var_insert_query
+
+        # Create temporary
+        temp_name = "__" + tablename + str(random.randint(1000, 9999))
```

Comment:
It feels like we shouldn't have to deal with creating connections + temp tables and stuff in the scraper classes.

I've been overriding the `_insert_query` method to customize only the query (SQL) and then letting the base class handle the connection

Reply:
Another thought -- would it possibly make things easier to split these into two classes, one for data and another for variables?

Then we can override the minimal set of methods 

If we are worried about sharing code in the `__init__` method or elsewhere I think we could probably create one of the classes first (e.g. the data one) and then have the other class subclass the first and override only what is necessary

Reply:
The reason that I ended up putting them into a single class is that the referenced variable has to exist in the `uscensus.acs_variables` table before we can insert data for that variable into the `uscensus.acs_data` -- This was a brute force way that allowed me to ensure that the right order of operations occurred.

Reply:
I assume that we can enforce something like this at the airflow level (but don't know how) -- I'm happy to break them back into two classes if we can ensure that the order of operations will be protected during any automated collection.

Reply:
Sigh... You're right. It is way cleaner broken into two classes...